bin/flightsubscriber/src/main.rs

```diff
@@ -1,42 +1,73 @@
-use arrow_flight::{encode::FlightDataEncoderBuilder, FlightClient, FlightDescriptor};
-use futures::StreamExt;
+use arrow_flight::{
+    decode::{DecodedPayload, FlightDataDecoder},
+    error::FlightError,
+    flight_service_client::FlightServiceClient,
+    FlightData, FlightDescriptor,
+};
+use futures::{stream, StreamExt};
 use tonic::transport::Channel;
+use tracing_subscriber::filter::Directive;
 
 /// Reads a Parquet file and sends it via DoPut to an Apache Arrow Flight endpoint.
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let _ = init_tracing();
+
     // Set up the Flight client
     let channel = Channel::from_static("http://localhost:50051")
         .connect()
         .await?;
-    let mut client = FlightClient::new(channel);
+    let mut client = FlightServiceClient::new(channel);
 
     let flight_descriptor = FlightDescriptor::new_path(vec!["test".to_string()]);
-    let flight_data_stream = FlightDataEncoderBuilder::new()
-        .with_flight_descriptor(Some(flight_descriptor))
-        .build(futures::stream::iter(
-            vec![].into_iter().map(Ok).collect::<Vec<_>>(),
-        ))
-        .map(Result::unwrap);
+    let subscription_request =
+        stream::iter(vec![FlightData::new().with_descriptor(flight_descriptor)].into_iter());
 
     println!("Subscribing to Apache Arrow Flight endpoint.");
-    let mut stream = client.do_exchange(flight_data_stream).await?;
+    let stream = client.do_exchange(subscription_request).await?;
+
+    let stream = stream.into_inner();
+
+    let mut flight_decoder = FlightDataDecoder::new(stream.map(|r| r.map_err(FlightError::Tonic)));
 
     loop {
-        let msg = stream.next().await;
+        let msg = flight_decoder.next().await;
         match msg {
-            Some(Ok(msg)) => {
-                println!("Received message: {msg:?}");
-            }
+            Some(Ok(msg)) => match msg.payload {
+                DecodedPayload::Schema(_) => {
+                    tracing::trace!("SCHEMA");
+                }
+                DecodedPayload::RecordBatch(batch) => {
+                    tracing::info!("RECORD BATCH: num_rows={}", batch.num_rows());
+                }
+                DecodedPayload::None => {
+                    tracing::trace!("NONE");
+                }
+            },
             Some(Err(e)) => {
-                println!("Error receiving message: {e:?}");
+                tracing::error!("Error receiving message: {e:?}");
             }
             None => {
-                println!("No more messages.");
+                tracing::info!("No more messages.");
                 break;
             }
         }
     }
 
     Ok(())
 }
+
+fn init_tracing() -> Result<(), tracing::subscriber::SetGlobalDefaultError> {
+    let filter = tracing_subscriber::EnvFilter::builder()
+        .with_default_directive("flightsubscriber".parse::<Directive>().unwrap())
```

Comment:
Define and return an error here instead of just using `unwrap`

Reply:
Probably just use `dyn std::error::Error` instead of `SetGlobalDefaultError` and `?` instead of `unwrap`
```
fn init_tracing() -> Result<(), Box<dyn std::error::Error>> {
    let filter = tracing_subscriber::EnvFilter::builder()
        .with_default_directive("flightsubscriber".parse::<Directive>()?)
```

Reply:
This is just a test utility.